### PR TITLE
Update testing chapter

### PR DESCRIPTION
**Pull request by danijelbuhin**
- Updated setup section
- Added naming convention
- Added events section
- Remove act usage
- Updated test examples

### CODE REVIEW

**Review by isBatak — APPROVED** [review]

File: React/Testing best practices.md [hunk 1]
```diff
@@ -237,78 +210,215 @@ Query priorities ([more info](https://testing-library.com/docs/guide-which-query
...
 describe("UserCard", () => {
-  it("Is rendering", () => {
+  it("should display the correct title", () => {
     const username = "Test User";
 
-    const { getByText } = render(<UserCard label={username} />);
+    render(<UserCard title={username} />);
 
-    expect(getByText(username)).toBeDefined();
+    expect(screen.(username)).toBeDefined();
```
Comment (kristijanbambir) [inline React/Testing best practices.md:320]:
```suggestion
    expect(screen.getByText(username)).toBeDefined();
```

As we talked in person, please add a paragraph on using screen over destructuring the render method return object.

File: React/Testing best practices.md [hunk 2]
```diff
@@ -237,78 +210,215 @@ Query priorities ([more info](https://testing-library.com/docs/guide-which-query
...
+  })
+```
+
+> **Note:** This is a shortened example of this concept, you can refer to the standup video section for more info:
+[Next.js testing - Testing shared components](https://youtu.be/KfaFyB0uedk?t=1005)
+
+
+## Testing user events
+
+If any part of your test is preforming the state update, that action needs to be wrapped into `waitFor`.
```
Comment (kristijanbambir) [inline React/Testing best practices.md:384]:
```suggestion
If any part of your test is performing the state update, that action needs to be wrapped into `waitFor`.
```

File: React/Testing best practices.md [hunk 3]
```diff
@@ -237,78 +210,215 @@ Query priorities ([more info](https://testing-library.com/docs/guide-which-query
...
+
+You would be also warned for the bad testing practices by the react-testing-library linter plugin.
+
+For simulating an event, we use `@testing-library/user-event` library.
```
Comment (kristijanbambir) [inline React/Testing best practices.md:388]:
```suggestion
For simulating an event, we use the `@testing-library/user-event` library.
```

File: React/Testing best practices.md [hunk 4]
```diff
@@ -336,7 +446,7 @@ const UserPage: NextPage = () => {
 export default UserPage;
 ```
 
-It this page example we should test all three states of the page component. To do that we should mock `<ErrorPage>`, `<LoadingPage>` and `<UserTemplate>` and check if it's rendered based on fetcher response.
+In this page example we should test all three states of the page component. To do that we should mock `<ErrorPage>`, `<LoadingPage>` and `<UserTemplate>` and check if it's rendered based on fetcher response.
```
Comment (kristijanbambir) [inline React/Testing best practices.md:449]:
```suggestion
In this page example we should test all three states of the page component. To do that we should mock `<ErrorPage>`, `<LoadingPage>` and `<UserTemplate>` and check if it's rendered based on the fetcher response.
```

File: React/Testing best practices.md [hunk 5]
```diff
@@ -387,26 +495,20 @@ Because of testing `useSWR` behavior (which will re-render DOM after fetcher pro
...
 ### SWR testing
 
-Testing swr has a few known issues. One is cache persistency across tests. To fix that we need to add this:
-
-```tsx
-afterEach(async () => {
-  await waitFor(() => cache.clear());
-});
-```
+To use swr in your tests, add `SWRConfig` to your `AllProviders` mock with following setup (note: your cases could require more customization):
```
Comment (kristijanbambir) [inline React/Testing best practices.md:498]:
```suggestion
To use SWR in your tests, add `SWRConfig` to your `AllProviders` mock with the following setup (note: your cases could require more customization):
```

File: React/Testing best practices.md [hunk 6]
```diff
@@ -387,26 +495,20 @@ Because of testing `useSWR` behavior (which will re-render DOM after fetcher pro
...
-  <SWRConfig value={{ dedupingInterval: 0 }}>
+  <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
     <ChakraProvider theme={theme}>{children}</ChakraProvider>
   </SWRConfig>
 );
 ```
 
-More can be found in [this issue](https://github.com/vercel/swr/issues/781).
-[Additional info](https://medium.com/frontend-digest/using-testing-libary-with-useswr-f595919de2fd).
+With this setup, we are sure that each component we're testing will have it's own cache, so we don't have to manually clear cache before each test.
```
Comment (kristijanbambir) [inline React/Testing best practices.md:509]:
```suggestion
With this setup, we are sure that each component we're testing will have its own cache, so we don't have to manually clear the cache before each test.
```

**Review by kristijanbambir — CHANGES_REQUESTED** [review]

File: React/Testing best practices.md [hunk 7]
```diff
@@ -215,14 +215,19 @@ Based on [the Guiding Principles](https://testing-library.com/docs/guiding-princ
 
 **Avoid unnecessary "is rendering" test**
 
-Since we have no value in testing wether our component will correctly render, avoid writing this type of tests and focus on more valuable tests instead.
+Since we have no value in testing whether our component will correctly render, avoid writing these type of tests and focus on more valuable tests instead.
+
+**Avoid destructuring `render` result for querying/finding elements**
+
+It is recommended that you avoid destructuring the `render(...)` result and use `screen` object instead.
+To see more info about [why you should use screen](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen), and other common mistakes in RTL usage, please refer to the [Kents](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) blog post.
```
Comment (kristijanbambir) [inline React/Testing best practices.md:223]:
```suggestion
To see more info about [why you should use screen](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen), and other common mistakes in RTL usage, please refer to [Kent's](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library) blog post.
```

**Review by kristijanbambir — APPROVED** [review]

**Pull request merged**